File: Valid Palindrome - Leetcode 125/Valid Palindrome - Leetcode 125.js
var isPalindrome = function(s) {
    let L = 0, R = s.length - 1;

    while (L < R) {
        while (L < R && !/[a-zA-Z0-9]/.test(s[L])) {
            L++;
        }
        while (L < R && !/[a-zA-Z0-9]/.test(s[R])) {
            R--;
        }
        if (L < R && s[L].toLowerCase() !== s[R].toLowerCase()) {
            return false;
        }
        L++;
        R--;
    }
    return true;
};

File: Valid Palindrome - Leetcode 125/Valid Palindrome - Leetcode 125.java
public class Solution {
    public boolean isPalindrome(String s) {
        int L = 0, R = s.length() - 1;

        while (L < R) {
            while (L < R && !Character.isLetterOrDigit(s.charAt(L))) {
                L++;
            }
            while (L < R && !Character.isLetterOrDigit(s.charAt(R))) {
                R--;
            }
            if (L < R && Character.toLowerCase(s.charAt(L)) != Character.toLowerCase(s.charAt(R))) {
                return false;
            }
            L++;
            R--;
        }
        return true;
    }
}
